factory/random.py

```diff
@@ -22,3 +24,6 @@ def reseed_random(seed):
     """Reseed factory.fuzzy's random generator."""
     r = random.Random(seed)
     set_random_state(r.getstate())
+
+    for locale in Faker._FAKER_REGISTRY:
+        Faker._FAKER_REGISTRY[locale].random.setstate(r.getstate())
```

Comment:
I was thinking of the code smell `inappropriate intimacy` but I don't know much about the project yet so you are probably a better judge. Could be overkill. Reference: https://sourcemaking.com/refactoring/smells/inappropriate-intimacy